File: frontend/pevu-client/api/order.ts
import api from './axios';

export type OrderStatus = 'PENDING' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED';

export type OrderItemRequest = {
  productId: number;
  quantity: number;
  price: number;
};

export type OrderRequest = {
  userId?: number;
  guestEmail?: string;
  guestName?: string;
  items: OrderItemRequest[];
  shippingAddress: string;
  total: number;
  paymentMethod: string;
};

export type OrderItemDto = {
  productId: number;
  quantity: number;
  price: number;
};

export type OrderResponse = {
  id: number;
  userId?: number;
  guestEmail?: string;
  guestName?: string;
  items: OrderItemDto[];
  shippingAddress: string;
  total: number;
  paymentMethod: string;
  status: OrderStatus;
  createdAt: string;
};

export async function placeOrder(userId: number, data: OrderRequest): Promise<OrderResponse> {
  const res = await api.post<OrderResponse>(`/order/${userId}`, data);
  return res.data;
}

export async function placeGuestOrder(data: OrderRequest): Promise<OrderResponse> {
  const res = await api.post<OrderResponse>(`/order/checkout/guest`, data);
  return res.data;
}

export async function getOrders(userId: number): Promise<OrderResponse[]> {
  const res = await api.get<OrderResponse[]>(`/order/${userId}`);
  return res.data;
}

export async function getOrder(orderId: number): Promise<OrderResponse> {
  const res = await api.get<OrderResponse>(`/order/details/${orderId}`);
  return res.data;
}

export async function cancelOrder(orderId: number, userId: number): Promise<OrderResponse> {
  const res = await api.put<OrderResponse>(`/order/${orderId}/cancel`, undefined, { params: { userId } });
  return res.data;
} 

File: backend/order-service/bin/main/com/pevu/orderservice/model/Order.kt
package com.pevu.orderservice.model

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "orders")
data class Order(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    val userId: Long? = null,
    val guestEmail: String? = null,
    val guestName: String? = null,
    @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], orphanRemoval = true)
    val items: List<OrderItem> = emptyList(),
    val shippingAddress: String,
    val total: Int,
    val paymentMethod: String,
    @Enumerated(EnumType.STRING)
    val status: OrderStatus = OrderStatus.PENDING,
    val createdAt: Instant = Instant.now()
)

@Entity
@Table(name = "order_items")
data class OrderItem(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
    val order: Order,
    val productId: Long,
    val quantity: Int,
    val price: Int
)

enum class OrderStatus { PENDING, SHIPPED, DELIVERED, CANCELLED } 